raft.c

```diff
@@ -934,7 +943,7 @@ static raft_node_id_t makeRandomNodeId(RedisRaftCtx *rr)
     do {
         RedisModule_GetRandomBytes((unsigned char *) &id, sizeof(id));
         id &= ~(1<<31);
-    } while (!id || (rr->raft && raft_get_node(rr->raft, id) != NULL));
+    } while (!id || (rr->raft && raft_get_node(rr->raft, id) != NULL) || hasNodeIdBeenUsed(rr, id));
```

Comment:
Am I missing something here? `makeRandomNodeId()` would always get called by a node that is not yet part of the cluster, so `hasNodeIdBeenUsed()` should always be `false`.

Reply:
Well, theoretically there could be a clash where the random id is equal to a previously used id, which we check for.
In practice the chance is low, but why not check for it?

Reply:
Per out last discussion, we'll keep this.